modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaSchedulerEntryImplConstructorCheck.java

```diff
@@ -0,0 +1,140 @@
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaConstructor;
+import com.liferay.source.formatter.parser.JavaParameter;
+import com.liferay.source.formatter.parser.JavaSignature;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaSchedulerEntryImplConstructorCheck
+	extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> extendedClassNames = javaClass.getExtendedClassNames();
+
+		if (!extendedClassNames.contains("SchedulerEntryImpl")) {
+			return content;
+		}
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaConstructor()) {
+				continue;
+			}
+
+			JavaConstructor javaConstructor = (JavaConstructor)childJavaTerm;
+
+			content = StringUtil.replace(
+				content, javaConstructor.getContent(),
+				_replaceConstructor(javaConstructor));
+		}
+
+		return content;
+	}
+
+	@Override
+	protected String[] getNewImports() {
+		return _imports.toArray(new String[0]);
+	}
+
+	private String _getNewSuperWithSchedulerEntryImpl(String parameterName) {
+		StringBundler sb = new StringBundler(7);
+
+		sb.append("super(");
+		sb.append(parameterName);
+		sb.append(".getEventListenerClass(), ");
+		sb.append(parameterName);
+		sb.append(".getTriggerConfiguration(), ");
+		sb.append(parameterName);
+		sb.append(".getDescription());");
+
+		return sb.toString();
+	}
+
+	private String _replaceConstructor(JavaConstructor javaConstructor) {
+		String constructorContent = javaConstructor.getContent();
+
+		Matcher matcher = _superPattern.matcher(constructorContent);
+
+		if (!matcher.find()) {
+			return StringUtil.replace(
+				constructorContent, "{\n",
+				"{\n\t\t" + _replaceSuper(javaConstructor));
+		}
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			matcher.group());
+
+		if (parameterList.isEmpty()) {
+			return StringUtil.replace(
+				constructorContent, matcher.group(),
+				_replaceSuper(javaConstructor));
+		}
+
+		JavaSignature javaSignature = javaConstructor.getSignature();
+
+		for (JavaParameter javaParameter : javaSignature.getParameters()) {
+			String parameterType = javaParameter.getParameterType();
+
+			if (parameterType.equals("Trigger")) {
+				_imports = new ArrayList<>();
+
+				_imports.add(
+					"com.liferay.portal.kernel.scheduler.TriggerConfiguration");
```

Comment:
Hi @michaelmdelcavalcanti, why we declare _imports as an array and redeclare if necessary with this value?